tide_core.module

```diff
@@ -244,3 +247,20 @@ function tide_core_editor_js_settings_alter(array &$settings) {
     $settings['editor']['formats'][$text_format_id]['editorSettings']['contentsCss'][] = '/' . drupal_get_path('module', 'tide_core') . '/css/ckeditor_overrides.css';
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function tide_core_menu_link_content_access(EntityInterface $menu_link_item, $operation, AccountInterface $account) {
+  $url = $menu_link_item->getUrlObject();
+  if (!$url->isExternal() && $url->isRouted() && $menu_link_item->isPublished()) {
+    $parameters = $url->getRouteParameters();
+    if (isset($parameters['node'])) {
+      $node = Node::load($parameters['node']);
+      if ($account->hasPermission('administer menu')) {
+        return AccessResult::neutral();
+      }
+      return $node->isPublished() ? AccessResult::neutral() : AccessResult::forbidden();
```

Comment:
isn't it going to return the `AccessResult::neutral()` if the node is published? What if it does not has the permission `administer menu` and it is published?

Reply:
this is a good point.